sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ColumnVector.java

```diff
@@ -69,14 +69,14 @@ public abstract class ColumnVector implements AutoCloseable {
   public abstract void close();
 
   /**
-   * Cleans up memory for this column vector if it's not writable. The column vector is not usable
-   * after this.
+   * Cleans up memory for this column vector if it's resources are freeable between batches.
+   * The column vector is not usable after this.
    *
-   * If this is a writable column vector, it is a no-op.
+   * If this is a writable column vector or constant column vector, it is a no-op.
    */
-  public void closeIfNotWritable() {
-    // By default, we just call close() for all column vectors. If a column vector is writable, it
-    // should override this method and do nothing.
+  public void closeIfFreeable() {
```

Comment:
Hmm `ColumnVector` is an API actually. Can we avoid direct renaming here

Reply:
The original PR which added this method https://github.com/apache/spark/commit/800faf0abfa368ad0a5ef1e0fa44b74dbaab724e was merged ~a month ago. is it too late to now change the method name?

If so, how do you suggest we go about this? I think `closeIfNotWritable` doesn't represent the purpose of the method well, so one option is to deprecate `closeIfNotWritable` and directly call `closeIfFreeable` inside it. This is pretty ugly, though, so I'm open to any suggestions

Reply:
ah okie dokie then should be fine

Reply:
Yea, it is a new API just added recently and not released yet.